tools/inference.py

```diff
@@ -60,10 +59,14 @@ def infer() -> None:
     extension = os.path.splitext(args.weight_path)[-1]
```

Comment:
Now that the paths are of type `Path`, we could maybe use `extension = args.weight_path.suffix` and remove `import os` from the import at the top of the module.